newlib/libc/include/machine/endian.h

```diff
@@ -53,18 +53,56 @@ __bswap64(__uint64_t _x)
 }
 #endif /* !__GNUC__ */
 
+/* byteorder(3) - simimlar to linux <arpa/inet.h> */
 #ifndef __machine_host_to_from_network_defined
 #if _BYTE_ORDER == _LITTLE_ENDIAN
 #define	__htonl(_x)	__bswap32(_x)
 #define	__htons(_x)	__bswap16(_x)
 #define	__ntohl(_x)	__bswap32(_x)
 #define	__ntohs(_x)	__bswap16(_x)
+#define	htonl(_x)	__bswap32(_x)
+#define	htons(_x)	__bswap16(_x)
+#define	ntohl(_x)	__bswap32(_x)
+#define	ntohs(_x)	__bswap16(_x)
 #else
 #define	__htonl(_x)	((__uint32_t)(_x))
 #define	__htons(_x)	((__uint16_t)(_x))
 #define	__ntohl(_x)	((__uint32_t)(_x))
 #define	__ntohs(_x)	((__uint16_t)(_x))
+#define	htonl(_x)	((__uint32_t)(_x))
+#define	htons(_x)	((__uint16_t)(_x))
+#define	ntohl(_x)	((__uint32_t)(_x))
+#define	ntohs(_x)	((__uint16_t)(_x))
 #endif
```

Comment:
I'd define the new macros in terms of the existing ones -- will be a shorter patch which is easier to review.

Reply:
You mean like #define htonl __htonl? Yes that's right. That is more beautiful. Do you want to make that quick change after merge, otherwise I'm happy to just tweak it a bit although I'm not at my computer for the next couple of days.

Reply:
Yup, `#define htonl(_x) __htonl(_x)` style would be awesome. I'm happy to wait for you to fix it up; no sense pulling and then immediately whacking it around again.

Reply:
Okay, what's the reason for using underscore in front of the variable x, when used in a macro like this?

Reply:
I'm afraid I don't recall the details on why, but as I recall, it avoids potential problems with the textual nature of the preprocessor and local variables using the same name. I'd treat it as style and just follow the local conventions in the file.

Reply:
Looks great; time to merge!